crates/common/validator/src/validator.rs

```diff
@@ -266,6 +272,104 @@ impl ValidatorService {
         Ok(())
     }
 
+    pub async fn perform_sync_duties(&mut self, slot: u64) -> anyhow::Result<()> {
+        let mut interval = {
+            let interval_start = Instant::now();
+            interval_at(
+                interval_start,
+                Duration::from_secs(network_spec().seconds_per_slot * 2 / INTERVALS_PER_SLOT),
+            )
+        };
+        interval.set_missed_tick_behavior(MissedTickBehavior::Burst);
+
+        let (aggregator_infos, non_aggregator_indicies) = self
+            .sync_committee_duties
+            .iter()
+            .try_fold::<_, _, Result<_, anyhow::Error>>(
+            (
+                Vec::with_capacity(self.sync_committee_duties.len()),
+                Vec::with_capacity(self.sync_committee_duties.len()),
+            ),
+            |(mut aggregator_infos, mut non_aggregator_indicies), duty| {
+                if let Some(keystore) = self.validator_index_to_keystore.get(&duty.validator_index)
+                {
+                    for &committee_index in &duty.validator_sync_committee_indices {
+                        let proof = get_sync_committee_selection_proof(
+                            slot,
+                            committee_index,
+                            &keystore.private_key,
+                        )
+                        .map_err(|err| anyhow!("Could not get selection proof: {err:?}"))?;
+                        if is_sync_committee_aggregator(&proof) {
+                            aggregator_infos.push((
+                                duty.validator_index,
+                                committee_index,
+                                proof,
+                                Arc::clone(keystore),
+                            ));
+                        } else {
+                            non_aggregator_indicies.push(duty.validator_index);
+                        }
+                    }
+                }
+                Ok((aggregator_infos, non_aggregator_indicies))
+            },
+        )?;
+
+        let client = self.beacon_api_client.clone();
+
+        tokio::try_join!(
+            self.submit_sync_committee(slot, non_aggregator_indicies.as_ref()),
+            async move {
+                interval.tick().await;
```

Comment:
we aren't spawning a tokio/green thread, so I am not sure what is the point of this try_join!